addons/sale/migrations/12.0.1.1/pre-migration.py

```diff
@@ -44,13 +44,18 @@ def fill_sale_order_line_sections(cr):
         WHERE sol.id = sub.id
         """,
     )
+    openupgrade.logged_query(
+        cr, """
+        ALTER TABLE sale_order_line ALTER COLUMN product_id DROP not null
+        """,
+    )
     openupgrade.logged_query(
         cr, """
         INSERT INTO sale_order_line (order_id, layout_category_id,
-            sequence, name, price_unit, product_uom_qty, customer_lead,
+            sequence, name, price_unit, product_uom, product_uom_qty, customer_lead,
             display_type, create_uid, create_date, write_uid, write_date)
         SELECT sol.order_id, sol.layout_category_id,
-            min(sol.sequence) -1 as sequence, max(slc.name), 0, 0, 0,
+            min(sol.sequence) -1 as sequence, max(slc.name), 0, 1, 0, 0,
```

Comment:
You are harcoding a UoM, which is not correct.

Reply:
In this case I think it will be better drop the non null constraint for product_uom too (like product_id), what do you think @pedrobaeza ?

Reply:
Is it required in v12? If not, drop it.

Reply:
is not required

Reply:
It's required V11 so when we are trying to add sections to sale order line we get blocked by the non-null constraint 
In V12 it's not required when display_type field is set